packages/core/src/logic/creators/createOperator.ts

```diff
@@ -21,8 +22,20 @@ export default function createOperator<
   return {
     type,
     allowedTypes,
-    apply() {
-      throw new Error('Not implemented');
+    apply(args: unknown[]) {
+      const argsType = typeof args[0];
+
+      if (!allowedTypes.includes(argsType as Type))
```

Comment:
suggestion(blocking): each arg can have its own type, so we should check them all